torch/_inductor/codecache.py

```diff
@@ -3034,11 +3145,17 @@ def generate_halide_async(cls, meta: HalideMeta, source_code: str, submit_fn=Non
                 )
             )
 
+        binding_types = [
+            arg.bindings_type() for arg in meta.argtypes if arg.alias_of is None
+        ]
+        if meta.is_cuda():
+            binding_types.append("uintptr_t")  # stream
         bindings_future = cls.load_pybinding_async(
-            [arg.bindings_type() for arg in meta.argtypes],
-            cls._codegen_glue(meta.argtypes, headerfile),
-            extra_flags=(libfile,),
+            binding_types,
+            cls._codegen_glue(meta, headerfile),
+            extra_flags=(libfile, cls.build_standalone_runtime()),
```

Comment:
I'm a little puzzled here. Previously we don't pass in the runtime lib explicitly when building halide kernels . Is a system wide copy of the runtime lib get used?

Reply:
Before every single kernel would get its own copy of the runtime.  The "no_runtime" flag that this PR adds requires a seperate runtime be linked.

Reply:
Got it. Make sense now!